gapis/resolve/memory.go

```diff
@@ -65,6 +68,16 @@ func filterTypedRanges(ranges []*service.TypedMemoryRange) []*service.TypedMemor
 				end = ranges[i].Range.Base + ranges[i].Range.Size
 			}
 
+			if newRanges[last].Range.Size != end-start {
```

Comment:
This condition is being used to filter duplicates. Once we get rid of the duplication, we can get rid of this and the above sort criterion.

Reply:
By the way, items that trigger both the OnRead and OnWrite methods (e.g. VkFramebuffer in VkCreateFramebuffer) will inherently be duplicated, so we probably can't get rid of this unless we make a distinction between the two kinds of ranges. However, this doesn't account for the duplication of entries in an array.